File: src/libs/ThirdParty.VBSuppliers/Properties/ExpediaRapid/SerializableClasses/Book/Phone.vb
Imports Newtonsoft.Json
Namespace ExpediaRapid.SerializableClasses.Book

    Public Class Phone

        <JsonProperty("country_code")>
        Public Property CountryCode As String

        <JsonProperty("area_code")>
        Public Property AreaCode As String

        <JsonProperty("number")>
        Public Property Number As String

        Public Sub New()
        End Sub

        Public Sub New(phone As String)
            If String.IsNullOrWhiteSpace(phone) Then Throw New ArgumentException("Phone number not supplied")

            phone = phone.TrimStart({"+"c, "0"c})

            Me.CountryCode = phone.Remove(2)
            phone = phone.TrimStart(Me.CountryCode.ToCharArray)

            Me.AreaCode = phone.Remove(2)
            Me.Number = phone.TrimStart(Me.AreaCode.ToCharArray)

        End Sub

    End Class

End Namespace

File: src/libs/iVectorOne.Suppliers/ExpediaRapid/SerializableClasses/Book/Phone.cs
namespace iVectorOne.Suppliers.ExpediaRapid.SerializableClasses.Book
{
    using System;
    using Newtonsoft.Json;

    public class Phone
    {

        [JsonProperty("country_code")]
        public string CountryCode { get; set; }

        [JsonProperty("area_code")]
        public string AreaCode { get; set; }

        [JsonProperty("number")]
        public string Number { get; set; }

        public Phone()
        {
        }

        public Phone(string phone)
        {
            if (string.IsNullOrWhiteSpace(phone))
                throw new ArgumentException("Phone number not supplied");

            phone = phone.TrimStart(new[] { '+', '0' });

            CountryCode = phone.Remove(2);
            phone = phone[2..];

            AreaCode = phone.Remove(2);
            Number = phone[2..];
        }
    }
}